app/uk/gov/hmrc/exports/repositories/SubmissionRepository.scala

```diff
@@ -51,56 +51,60 @@ class SubmissionRepository @Inject()(implicit mc: ReactiveMongoComponent, ec: Ex
-      case Some(result) => logDatabaseResult(result)
-      case _            => false
+    val finder = Json.obj(
+      "_id" -> Json.toJsFieldJsValueWrapper(submission.id)(objectIdFormats),
+      "conversationId" -> submission.conversationId
+    )
+    val modifier = Json.obj("$set" -> Json.obj("mrn" -> submission.mrn, "status" -> submission.status))
+
+    findAndUpdate(finder, modifier, fetchNewObject = true).map {
+      case result if result.value.isEmpty =>
+        result.lastError.foreach(_.err.foreach(logDatabaseResult))
+        false
+      case _ =>
+        true
     }
   }
 
-  def updateMrnAndStatus(eori: String, convId: String, mrn: String, status: Option[String]): Future[Boolean] =
-    if (status.isDefined) {
-      val finder = BSONDocument("eori" -> eori, "conversationId" -> convId)
-      val modifier = BSONDocument("$set" -> BSONDocument("mrn" -> mrn, "status" -> status.get))
-
-      atomicUpdate(finder, modifier).map {
-        case Some(result) => logDatabaseResult(result)
-        case _            => false
+  // TODO: Get rid of Option[_]
+  def updateMrnAndStatus(eori: String, convId: String, newMrn: String, newStatus: Option[String]): Future[Boolean] =
+    if (newStatus.isDefined) {
+      val finder = Json.obj("eori" -> eori, "conversationId" -> convId)
+      val modifier = Json.obj("$set" -> Json.obj("mrn" -> newMrn, "status" -> newStatus.get))
+
+      findAndUpdate(finder, modifier, fetchNewObject = true).map {
+        case result if result.value.isEmpty =>
+          result.lastError.foreach(_.err.foreach(logDatabaseResult))
+          false
+        case _ =>
+          true
       }
     } else Future.successful(false)
 
   def cancelDeclaration(eori: String, mrn: String): Future[CancellationStatus] = {
-    val finder = BSONDocument("eori" -> eori, "mrn" -> mrn)
+    val finder = Json.obj("eori" -> eori, "mrn" -> mrn)
+    val modifier =
+      Json.obj("$set" -> Json.obj("status" -> RequestedCancellation.toString, "isCancellationRequested" -> true))
 
-    val modifier = BSONDocument(
-      "$set" -> BSONDocument("status" -> RequestedCancellation.toString, "isCancellationRequested" -> true)
-    )
-
-    find("eori" -> JsString(eori), "mrn" -> JsString(mrn))
-      .map(_.headOption) flatMap {
+    find("eori" -> JsString(eori), "mrn" -> JsString(mrn)).map(_.headOption).flatMap {
       case Some(submission) if submission.isCancellationRequested =>
         Future.successful(CancellationRequestExists)
       case Some(_) =>
-        atomicUpdate(finder, modifier).map {
-          case Some(result) if logDatabaseResult(result) => CancellationRequested
-          case _                                         => MissingDeclaration
+        findAndUpdate(finder, modifier, fetchNewObject = true).map {
+          case result if result.lastError.isDefined && result.lastError.get.err.isDefined =>
+            logDatabaseResult(result.lastError.get.err.getOrElse("No error message found"))
```

Comment:
As we discussed yesterday, you're checking that err in last error isDefined `result.lastError.get.err.isDefined` so this `getOrElse` will not be used